provider/postgis/postgis.go

```diff
@@ -201,38 +207,43 @@ func BuildDBConfig(cs string) (*pgxpool.Config, error) {
 		"application_name":              "tegola",
 	}
 
-	var hasInit bool
+	var hstore hstoreOID
 	dbconfig.AfterConnect = func(ctx context.Context, conn *pgx.Conn) error {
 		// The AfterConnect call runs everytime a new connection is acquired,
 		// including everytime the connection pool expands. The hstore OID
-		// lookup should only be run once
-		if !hasInit {
-			// as extension hstore does not have a fixed OID we have to get it on connect
-			var hstoreOID uint32
+		// is not constant, so we lookup the OID once per provider and store it.
+		// Extensions have to be registered for every new connection.
+
+		if !hstore.hasInit {
 			row := conn.QueryRow(ctx, "SELECT oid FROM pg_type WHERE typname = 'hstore';")
-			if err = row.Scan(&hstoreOID); err != nil {
+			if err = row.Scan(&hstore.OID); err != nil {
 				switch {
 				case strings.Contains(err.Error(), "no rows in result set"):
```

Comment:
We could probably clean up this check by switching to:

```go
case errors.Is(pgx.ErrNoRows):
```

https://pkg.go.dev/github.com/jackc/pgx/v4#ErrNoRows

Reply:
✅